ResourceStatusSystem/PolicySystem/StateMachine.py

```diff
@@ -116,32 +116,33 @@ def levelOfState( self, state ):
       return -1
     return self.states[ state ].level
 
-  def setState( self, state ):
+  def setState(self, candidateState):
+    """ Makes sure the state is either None or known to the machine, and that it is a valid state to move into.
+        Final states are also checked.
+        This is a re-definition of original one that wasn't making these checks
     """
-      Makes sure the state is either None or known to the machine
 
-    examples:
-      >>> sm0.setState( None )[ 'OK' ]
-          True
-      >>> sm0.setState( 'Nirvana' )[ 'OK' ]
-          True
-      >>> sm0.setState( 'AnotherState' )[ 'OK' ]
-          False
-
-    :param state: state which will be set as current state of the StateMachine
-    :type state: None or str
-    :return: S_OK || S_ERROR
```

Comment:
Why did you remove the lines above?

Reply:
by mistake...
Fixed:
8b0ea690e093080f21879899ec80301b0326996e